contracts/usul/BaseStrategy.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../interfaces/IFractalUsul.sol";
+import "./IBaseStrategy.sol";
+import "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+abstract contract BaseStrategy is
+    OwnableUpgradeable,
+    FactoryFriendly,
+    IBaseStrategy
+{
+    event UsulSet(address indexed newUsul);
+    event StrategySetup(address indexed UsulModule, address indexed owner);
+
+    IFractalUsul public usulModule;
+
+    modifier onlyUsul() {
+        require(
+            msg.sender == address(usulModule),
+            "Only callable by Usul module"
+        );
+        _;
+    }
+
+    /// @notice Sets the address of the Usul contract, only callable by owner
+    /// @param _usulModule The address of the Usul module
+    function setUsul(address _usulModule) external onlyOwner {
```

Comment:
I referenced this is a previous comment, but yeah, we should be consistent with underscore / no underscore for parameters throughout the code

Reply:
Hmm well in this case, `usulModule` would collide with the global variable name. We could go with underscores on all function args tho?

Reply:
yeah, going with underscores for all of them seems like a Solidity standard, so makes sense